### PR TITLE
[Mentorship] Updates for the 2022 Swift Mentorship Program.

### PR DESCRIPTION
**Pull request by hborla**
Update the community for the 2022 offering of the Swift Mentorship Program with a blog post and updates to Swift.org/mentorship.

### CODE REVIEW

File: _data/authors.yml [hunk 1]
```diff
@@ -268,4 +268,10 @@ diversity-work-group:
...
   twitter: 0xTim
   gravatar: c8678b78a2c1c113302ce686f13435d1
-  about: "Tim Condon sits on the SSWG (Swift Server Work Group) and is part of the Vapor Core Team"
\ No newline at end of file
+  about: "Tim Condon sits on the SSWG (Swift Server Work Group) and is part of the Vapor Core Team"
+
+devanshimodha:
+  name: Devanshi Modha
+  github: devanshimodha
+  twitter: devanshimodha
```
Comment (shahmishal) [inline _data/authors.yml:276]: It's missing `gravatar`, if they are interested in providing image.

File: _posts/2022-05-19-mentorship-2022.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
```
Comment (shahmishal) [inline _posts/2022-05-19-mentorship-2022.md:2]:
```suggestion
layout: post
published: true
```

File: _data/authors.yml (hunk 1 above)
Comment (hborla) [reply _data/authors.yml:276]: Ah! @devanshimodha would you like to provide an image, e.g. the one you use for the Swift Forums and Twitter?

Comment (hborla) [reply _data/authors.yml:276]: (We can also add this in a separate PR. In fact, I'm missing an image too, as is @krstnfx)

**Review by shahmishal — APPROVED** [review]
LGTM!

**Review by tkremenek — APPROVED** [review]

**Pull request merged**